aten/src/ATen/native/cuda/ForeachReduceOp.cu

```diff
@@ -117,10 +117,16 @@ std::vector<Tensor> foreach_tensor_norm_cuda(TensorList tensors, const Scalar& o
       const auto scalar_type = t.scalar_type();
       return at::isIntegralType(scalar_type, /*includeBool*/true) || at::isComplexType(scalar_type);
   });
+  const bool is_same_dtype = dtype.has_value() ? std::any_of(tensors.begin(), tensors.end(), [&](const auto & t) {
```

Comment:
Is this not already checked in check_foreach_api_restrictions?

Should this be checked there instead?

Reply:
Sorry perhaps a naming issue, but `is_same_dtype` checks that all tensors are of the same dtype as the specified `dtype` passed in argument (if specified). Typically you can run `torch.linalg.vector_norm(..., dtype=torch.bfloat16)`. It doesn't just check that all tensors are of the same dtype.